src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/schema/NumberCoreTest.kt

```diff
@@ -0,0 +1,141 @@
+                // start with +
+                row("+1", 1),
+                row("+1223344", 1223344),
+                row("+12.23344", 12.23344),
+                row("+0.23344", 0.23344),
+                row("+0", 0),
+                // leading zero
+                row("03", 3),
+                row("03.67", 3.67),
+            )
+        ) { string: String, value: Any ->
+            loader.loadOne(string) shouldBe value
+        }
+    }
+
+    test("all strings which WERE integers or doubles in YAML 1.1") {
+        forAll(
+            table(
+                headers("string", "value"),
+                row("12:10:02", "12:10:02"),
+                row("0b1010", "0b1010"),
+                row("1_000", "1_000"),
+                row("1_000.5", "1_000.5"),
+                row("-0xFF", "-0xFF"),
+                row("+0xFF", "+0xFF"),
+                row("+0o123", "+0o123"),
+                row("-0o123", "-0o123"),
+                row("! 3.6", "3.6"),
+                row("! 3", "3"),
+            )
+        ) { string: String, value: Any ->
+            loader.loadOne(string) shouldBe value
+        }
+    }
+
+    test("all doubles which are defined in the core schema & JSON") {
+        forAll(
+            table(
+                headers("string", "value"),
+                row("-1.345", -1.345),
+                row("0.0", 0.0),
+                row("-0.0", 0.0),
+                row("0.123", 0.123),
+                row("1.23e-6", 1.23E-6),
+                row("1.23e+6", 1.23E6),
+                row("1.23e6", 1.23E6),
+                row("1.23E6", 1.23E6),
+                row("-1.23e6", -1.23E6),
+                row("1000.25", 1000.25),
+                row("9000.0", 9000.0),
+                row("1.", 1.0),
+            )
+        ) { string: String, value: Any ->
+            loader.loadOne(string) shouldBe value
+        }
+    }
+
+    test("parse special doubles which are defined in the core schema") {
+        forAll(
+            table(
+                headers("string", "value"),
+                row(".inf", Double.POSITIVE_INFINITY),
+                row(".Inf", Double.POSITIVE_INFINITY),
+                row(".INF", Double.POSITIVE_INFINITY),
+                row("-.inf", Double.NEGATIVE_INFINITY),
+                row("-.Inf", Double.NEGATIVE_INFINITY),
+                row("-.INF", Double.NEGATIVE_INFINITY),
+                row(".nan", Double.NaN),
+                row(".NaN", Double.NaN),
+                row(".NAN", Double.NaN),
+            )
+        ) { string: String, value: Any ->
+            loader.loadOne(string)!! shouldBeEqual value
+        }
+    }
+
+    test("dump special doubles which are defined in the core schema") {
+        val dumper = Dump(DumpSettings.builder().setSchema(CoreSchema()).build())
+        forAll(
+            table(
+                headers("value", "string"),
+                // There's a different behavior for JS, and it's a bug.
```

Comment:
I think we need to add FIXME here (it is hard to search for something by a plain comment)

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/525/commits/ff6d0edbd0fae80a83fec72a19517bb65d2e9595